### PR TITLE
feat: Neo Session Launcher Tour Guide via 'antd tour'

### LINKED ISSUE #2260: Let's provide a tour guide for neo session launcher
### Main idea

With the introduction of neo session launcher, there are a lot of changes from the existing UI. Let's apply a tour guide to Neo session laucnher to help users understand. 

### Alternative ideas

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ironAiken2**
### Feature:
- set localstorage state to control tour needs to open or not.
- Provide Tour Guide for `Neo Session Launcher`.
- Controls the current step as a state, resetting it to the initial step when the sectional step changes.
- To access elements in a child component, used `data-test-id`.
- I kept the data structure for tour steps (for the possibility of adding other tour steps in the future)

### Request Changes
 [Applied]
- support i18n
- Resolve TypeError 
- use className or data-id instead of using ref

image

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
It's better to show the start step in the center using `target={null}` https://ant.design/components/tour#tour-demo-placement

**Comment by ironAiken2** [comment]
[* Show the start step in the center of screen](https://github.com/lablup/backend.ai-webui/pull/2284#issuecomment-2029260860) - applied

**Comment by yomybaby** [comment]
Please resolve the conflicts @ironAiken2

**Review by yomybaby — CHANGES_REQUESTED** [review]
There are too many steps in the tour. 
Let's keep only a few key steps and remove the rest. Let's split it into two tours: 
- Tour 1: In the first step, welcome to Neo Session Launcher, move to the right step, move to the bottom button. 
- Tour 2: Only display if there is a validation error in the final step. 
  - If there is a validation error, guide to press the modify button to correct it. 
  
And let's refactoring
- use `data-test-id` instead of using class names or refs. 
- After remaining only 2 tours, unnecessary refs can be deleted. 
- Instead of directly accessing localStorage in settings, use useBAISettingUserState. 
- Give a more specific name to the setting, indicating that it is for determining whether the tour has been completed. For example, (has_toured_neo_session_launcher_introduction):

**Comment by ironAiken2** [comment]
* **#2284** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2284?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ironAiken2** [comment]
Applied

> There are too many steps in the tour. 
> Let's keep only a few key steps and remove the rest. Let's split it into two tours: 
> - Tour 1: In the first step, welcome to Neo Session Launcher, move to the right step, move to the bottom button. 
> - Tour 2: Only display if there is a validation error in the final step. 
>   - If there is a validation error, guide to press the modify button to correct it. 
>   
> And let's refactoring
> - use `data-test-id` instead of using class names or refs. 
> - After remaining only 2 tours, unnecessary refs can be deleted. 
> - Instead of directly accessing localStorage in settings, use useBAISettingUserState. 
> - Give a more specific name to the setting, indicating that it is for determining whether the tour has been completed. For example, (has_toured_neo_session_launcher_introduction):
> 
>

File: react/src/pages/SessionLauncherPage.tsx [hunk 1]
```diff
@@ -1082,6 +1214,13 @@ const SessionLauncherPage = () => {
...
                   <>
                     <BAICard
+                      data-test-id={
+                        form.getFieldError('sessionName').length > 0 ||
+                        form.getFieldError(['batch', 'command']).length > 0 ||
+                        form.getFieldError(['batch', 'scheduleDate']).length > 0
+                          ? 'neo-session-launcher-tour-validation-error-feild'
+                          : undefined
+                      }
                       title={t('session.launcher.SessionType')}
```
Comment (yomybaby) [inline react/src/pages/SessionLauncherPage.tsx]: In this case, you can set the class name of `Card` in `BAICard` based on the `status`. Then, you won't need to set `data-test-id` with duplicated logic. Also you can query for the target of the tour step directly using calss name.
Before I told you that class name for tour purporse only is not good. However, this is a slightly different case. In this case, the class name is determined by the `status`.

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/pages/SessionLauncherPage.tsx [hunk 2]
```diff
@@ -567,6 +590,117 @@ const SessionLauncherPage = () => {
...
+          '필수 입력 필드에 값을 입력하지 않거나 잘못된 값을 입력한 경우, 해당 필드에 오류 메시지가 표시됩니다.',
+        target: () =>
+          document.querySelector(
+            '[data-test-id="neo-session-launcher-tour-validation-error-feild"]',
+          ) as HTMLElement,
+      },
+      {
+        title: '입력 필드 검증 오류',
+        description:
+          '해당 오류 필드에 대해, 우측 상단의 수정 버튼으로 바로 이동하여 수정할 수 있습니다. ',
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
          '해당 오류 필드에 대해, 우측 상단의 수정 버튼으로 바로 이동하여 수정할 수 있습니다.',
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 3]
```diff
@@ -567,6 +590,117 @@ const SessionLauncherPage = () => {
...
+  };
+  const onTourPrev = () => {
+    setCurrentTourStep((prev) => prev - 1);
+  };
+
+  const sectionSteps: { [key: string]: TourStepProps[] } = {
+    sessionType: [
+      {
+        title: 'Welcome!',
+        description: '새로운 NEO 세션 런처에 오신걸 환영합니다!.',
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
        description: '새로운 NEO 세션 런처에 오신 것을 환영합니다!',
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 4]
```diff
@@ -567,6 +590,117 @@ const SessionLauncherPage = () => {
...
+        target: null,
+      },
+      {
+        title: '세션 실행 단계',
+        description:
+          '우측의 단계 표시를 통해 현재 단계를 확인할 수 있습니다. 특정 단계를 클릭하여 이동할 수 있습니다.',
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
          '우측에 있는 단계 표시에서 현재 단계를 확인할 수 있습니다. 특정 단계를 클릭하여 이동할 수 있습니다.',
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 5]
```diff
@@ -567,6 +590,117 @@ const SessionLauncherPage = () => {
...
+        target: () =>
+          document.querySelector(
+            '[data-test-id="neo-session-launcher-tour-step"]',
+          ) as HTMLElement,
+      },
+      {
+        title: '단계 이동',
+        description:
+          'Next 버튼을 통하여 다음 단계로 이동하거나 review 단계로 건너뛸 수 있습니다.',
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
          '다음 버튼을 클릭하여 다음 단계로 이동하거나 검토로 건너뛸 수 있습니다.',
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 6]
```diff
@@ -567,6 +590,117 @@ const SessionLauncherPage = () => {
...
+            '[data-test-id="neo-session-launcher-tour-step-navigation"]',
+          ) as HTMLElement,
+      },
+    ],
+  };
+  console.log(
+    document.querySelector(
+      '[data-test-id="neo-session-launcher-tour-validation-error-feild"]',
+    ),
+  );
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
1. dark mode support is needed. (highlighting area)
image
2. support i18n
3. I got TypeError. Please recheck the id.
image

**Review by yomybaby — APPROVED** [review]
LGTM

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**